### DISCUSSION TITLE
Is token login /3rd party auth possible with the horizon theme?

### DISCUSSION THREAD

**Discussion by cheningD** [swellstores/community#61] [Help] [answered]
I am trying to replace the horizon theme login with a passwordless option. 
The docs mention that token login can be used with swell-node / swell-js to integrate a 3rd party auth provider. (https://developers.swell.is/storefronts/token-login) 

However, even when login is successful with swell-js/swell-node, it does not produce a sessionToken that is valid for the horizon theme (which uses the GraphQL Library)

Is there an easy way to allow token login for the horizon theme?

**Comment by rodrigoddalmeida** [comment, marked as answer]
You should be able to use the generated `password_token` by changing the `loginAccount` mutation to:

```graphql
mutation login($email: String, $passwordToken: String) {
  loginAccount(email: $email, passwordToken: $passwordToken) {
    success
  }
}
```

**Reply by cheningD** [reply]
This worked, Thank you! I ended up creating a new mutation so I could so the original would continue to work as intended.
